### PR TITLE
Add periodic-ipi-deprovision job

### PR DESCRIPTION
**Pull request by hongkailiu**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -34,7 +34,7 @@ applyTemplate:
 	oc process -f $(WHAT) | oc apply -f -
 .PHONY: applyTemplate
 
-postsubmit-update: prow-services origin-release libpod prow-monitoring build-dashboards-validation-image cincinnati
```
Comment (petr-muller) [inline Makefile]: Let's stop adding stuff to legacy places. If you add everything somewhere under `core-services`, you won't need to put anything to Makefile.

File: ci-operator/jobs/infra-periodics.yaml [hunk 2]
```diff
@@ -232,6 +232,36 @@ periodics:
     - configMap:
         name: labels
       name: config
+- agent: kubernetes
+  labels:
+    ci.openshift.io/role: infra
+  cron: ""
+  decorate: true
+  decoration_config:
+    timeout: 28800000000000 # 8h
```
Comment (petr-muller) [inline ci-operator/jobs/infra-periodics.yaml]: I believe we can already put "8h" here?

File: ci-operator/jobs/infra-periodics.yaml [hunk 3]
```diff
@@ -232,6 +232,36 @@ periodics:
...
+  interval: 5m
+  name: periodic-ipi-deprovision
+  ### did not find this. Should it be under `decoration_config`
```
Comment (petr-muller) [inline ci-operator/jobs/infra-periodics.yaml]: I believe it should.

File: projects/origin-release/ipi_deprovision/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,10 @@
+FROM registry.svc.ci.openshift.org/openshift/hive-v4.0:hive
```
Comment (petr-muller) [inline projects/origin-release/ipi_deprovision/Dockerfile]: This file should be somewhere under `core-services`, or even better, in the `ci-tools` repo.

File: projects/origin-release/ipi_deprovision/ipi-deprovision.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
+apiVersion: v1
```
Comment (petr-muller) [inline projects/origin-release/ipi_deprovision/ipi-deprovision.yaml]: This file should be somewhere under `core-services`, or even not needed if we build the image with `ci-operator`.

File: projects/origin-release/ipi_deprovision/run.sh [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
```
Comment (petr-muller) [inline projects/origin-release/ipi_deprovision/run.sh]: This file should be somewhere under `core-services`, or even better, in `ci-tools` repo.

Comment (hongkailiu) [reply projects/origin-release/ipi_deprovision/run.sh]: https://github.com/openshift/ci-tools/pull/26

File: projects/origin-release/ipi_deprovision/ipi-deprovision.yaml (hunk 5 above)
Comment (hongkailiu) [reply projects/origin-release/ipi_deprovision/ipi-deprovision.yaml]: https://github.com/openshift/ci-tools/pull/26

File: projects/origin-release/ipi_deprovision/Dockerfile (hunk 4 above)
Comment (hongkailiu) [reply projects/origin-release/ipi_deprovision/Dockerfile]: https://github.com/openshift/ci-tools/pull/26

File: projects/origin-release/ipi_deprovision/ipi-deprovision.yaml (hunk 5 above)
Comment (hongkailiu) [reply projects/origin-release/ipi_deprovision/ipi-deprovision.yaml]: https://github.com/openshift/ci-tools/pull/26

File: projects/origin-release/ipi_deprovision/Dockerfile (hunk 4 above)
Comment (hongkailiu) [reply projects/origin-release/ipi_deprovision/Dockerfile]: https://github.com/openshift/ci-tools/pull/26

File: ci-operator/config/openshift/ci-tools/openshift-ci-tools-master.yaml [hunk 7]
```diff
@@ -83,6 +88,14 @@ images:
       - destination_dir: .
         source_path: /go/bin/determinize-ci-operator
   to: determinize-ci-operator
+- context_dir: images/ipi_deprovision/
+  from: hive
+  inputs:
+    bin:
+      paths:
+      - destination_dir: .
+        source_path: /go/bin/ci-operator
```
Comment (hongkailiu) [inline ci-operator/config/openshift/ci-tools/openshift-ci-tools-master.yaml]: Need to be updated.

File: ci-operator/jobs/infra-periodics.yaml (hunk 2 above)
Comment (hongkailiu) [reply ci-operator/jobs/infra-periodics.yaml]: done.

File: ci-operator/jobs/infra-periodics.yaml (hunk 3 above)
Comment (hongkailiu) [reply ci-operator/jobs/infra-periodics.yaml]: removed `skip_cloning: true`

File: ci-operator/config/openshift/ci-tools/openshift-ci-tools-master.yaml [hunk 8]
```diff
@@ -9,6 +9,11 @@ base_images:
     name: release
     namespace: openshift
     tag: golang-1.11
+  hive:
+    cluster: https://api.ci.openshift.org
```
Comment (petr-muller) [inline ci-operator/config/openshift/ci-tools/openshift-ci-tools-master.yaml]: `cluster` does not need to be there (ideally, remove it also from the other two items)

File: ci-operator/config/openshift/ci-tools/openshift-ci-tools-master.yaml [hunk 9]
```diff
@@ -83,6 +88,14 @@ images:
       - destination_dir: .
         source_path: /go/bin/determinize-ci-operator
   to: determinize-ci-operator
+- context_dir: images/ipi-deprovision/
```
Comment (petr-muller) [inline ci-operator/config/openshift/ci-tools/openshift-ci-tools-master.yaml]: I think this stanza needs to be simply:

```
- dockerfile_path: images/ipi-deprovision/Dockerfile
  from: hive
  to: ipi-provisioner
```

(provided https://github.com/openshift/ci-tools/pull/26#discussion_r305897546 is addressed)

File: ci-operator/config/openshift/ci-tools/openshift-ci-tools-master.yaml (hunk 8 above)
Comment (hongkailiu) [reply ci-operator/config/openshift/ci-tools/openshift-ci-tools-master.yaml]: Done.

File: ci-operator/config/openshift/ci-tools/openshift-ci-tools-master.yaml (hunk 9 above)
Comment (hongkailiu) [reply ci-operator/config/openshift/ci-tools/openshift-ci-tools-master.yaml]: done.

**Pull request merged**